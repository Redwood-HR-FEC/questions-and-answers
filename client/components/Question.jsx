import React from 'react';
import styled from 'styled-components';

const QuestionWrapper = styled.div`
  margin-bottom: 10px!important;
`;

const ATag = styled.a`
  color:rgb(0, 102, 192);
  text-decoration-line:none;
  text-decoration-style:solid;

  &:hover {
    color:rgb(196, 85, 0);
    text-decoration-line:underline;
    text-decoration-style:solid;
  }
`;

const Question = ({ question, url }) => {
  return (
    <QuestionWrapper>
      <ATag href={url}>
        { question }
        ?
      </ATag>
    </QuestionWrapper>
  );
};

export default Question;
