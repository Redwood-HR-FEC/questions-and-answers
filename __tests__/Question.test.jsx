import React from 'react';
import { shallow } from 'enzyme';
import Question from '../client/components/Question';

describe('Unit test for Question component', () => {
  const sampleQuestion = 'tan Center Kansas Comoro Franc Developer sky blue Wooden open-source Designer Lead';
  const sampleURL = 'http://larissa.name';

  test('should render questions', () => {
    const wrapper = shallow(
      <Question
        question={sampleQuestion}
        url={sampleURL}
      />,
    );
    expect(wrapper).toExist();
  });

  it("renders correctly", () => {
    const wrapper = shallow(
      <Question
        question={sampleQuestion}
        url={sampleURL}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('Expect a tag to render the correct question', () => {
    const wrapper = mount(
      <Question
        question={sampleQuestion}
        url={sampleURL}
      />,
    );
    const value = wrapper.find('a').text();
    expect(value).toEqual('tan Center Kansas Comoro Franc Developer sky blue Wooden open-source Designer Lead?');
  });

  it ('accepts question and URL props', () => {
    const wrapper = mount(
      <Question
        question={sampleQuestion}
        url={sampleURL}
      />,
    );
    expect(wrapper.props().question).toEqual(sampleQuestion);
    expect(wrapper.props().url).toEqual(sampleURL);
  });
});
