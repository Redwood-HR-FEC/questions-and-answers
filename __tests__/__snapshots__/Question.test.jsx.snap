// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit test for Question component renders correctly 1`] = `
<Question__QuestionWrapper>
  <Question__ATag
    href="http://larissa.name"
  >
    tan Center Kansas Comoro Franc Developer sky blue Wooden open-source Designer Lead
    ?
  </Question__ATag>
</Question__QuestionWrapper>
`;
